skore/src/skore/sklearn/_estimator/base.py

```diff
@@ -0,0 +1,143 @@
+                filtered_methods.append((name, method))
+        return filtered_methods
+
+    def _sort_methods_for_help(self, methods):
+        """Sort methods for help display."""
+        return sorted(methods)
+
+    def _format_method_name(self, name):
+        """Format method name for display."""
+        return f"{name}(...)"
+
+    def _get_method_description(self, method):
+        """Get the description for a method."""
+        return (
+            method.__doc__.split("\n")[0]
+            if method.__doc__
+            else "No description available"
+        )
+
+    def _create_help_panel(self):
+        """Create the help panel."""
+        return Panel(
+            self._create_help_tree(),
+            title=self._get_help_panel_title(),
+            expand=False,
+            border_style="orange1",
+        )
+
+    def help(self):
+        """Display available methods using rich."""
+        from skore import console  # avoid circular import
+
+        console.print(self._create_help_panel())
+
+    def __repr__(self):
+        """Return a string representation using rich."""
+        console = Console(file=StringIO(), force_terminal=False)
+        console.print(self._create_help_panel())
+        return console.file.getvalue()
+
+
+class _BaseAccessor(_HelpMixin):
+    """Base class for all accessors."""
+
+    def __init__(self, parent, icon):
+        self._parent = parent
+        self._icon = icon
+
+    def _get_help_panel_title(self):
+        name = self.__class__.__name__.replace("_", "").replace("Accessor", "").lower()
+        return f"{self._icon} Available {name} methods"
+
+    def _create_help_tree(self):
+        """Create a rich Tree with the available methods."""
+        tree = Tree(self._get_help_tree_title())
+
+        methods = self._get_methods_for_help()
+        methods = self._sort_methods_for_help(methods)
+
+        for name, method in methods:
+            displayed_name = self._format_method_name(name)
+            description = self._get_method_description(method)
+            tree.add(f".{displayed_name}".ljust(26) + f" - {description}")
+
+        return tree
+
+    def _get_X_y_and_data_source_hash(self, *, data_source, X=None, y=None):
+        """Get the requested dataset and mention if we should hash before caching.
+
+        Parameters
+        ----------
+        data_source : {"test", "train", "X_y"}, default="test"
+            The data source to use.
+
+            - "test" : use the test set provided when creating the reporter.
+            - "train" : use the train set provided when creating the reporter.
+            - "X_y" : use the provided `X` and `y` to compute the metric.
+
+        Returns
+        -------
+        X : array-like of shape (n_samples, n_features)
+            The requested dataset.
+
+        y : array-like of shape (n_samples,)
+            The requested dataset.
+
+        data_source_hash : int or None
```

Comment:
Out-of-sync with arguments

Reply:
Nop. I return either a int hash or None.

Reply:
Ah you mean in the parameters above.